### PR TITLE
spectrum-mpi: external support, compiler detection

### PR DESCRIPTION
**Pull request by rblake-llnl**
This is the easiest MPI to add as it has no variants.  Tested on LLNL Lassen.

The selection criteria for finding the correct compiler was taken with advice from @becker33.

### CODE REVIEW

File: var/spack/repos/builtin/packages/spectrum-mpi/package.py [hunk 1]
```diff
@@ -12,6 +13,78 @@ class SpectrumMpi(Package):
...
+                    if compiler_name in siblings:
+                        # Get the real name of the compiler
+                        full_exe = os.path.join(dirname, compiler_name)
+                        host_exe = get_host_compiler(full_exe)
+                        if host_exe:
+                            compilers_found[compiler_class] = host_exe
+                if compilers_found:
+                    break
+            if compilers_found:
+                print(compilers_found)
```
Comment (becker33) [inline var/spack/repos/builtin/packages/spectrum-mpi/package.py]: This looks like a leftover from debugging.

File: var/spack/repos/builtin/packages/spectrum-mpi/package.py [hunk 2]
```diff
@@ -12,6 +13,78 @@ class SpectrumMpi(Package):
...
+                compiler_spec = get_spack_compiler_spec(compilers_found)
+                if compiler_spec:
+                    variant = "%" + str(compiler_spec)
+                else:
+                    variant = ''
+                results.append((variant, {'compilers': compilers_found}))
```
Comment (becker33) [inline var/spack/repos/builtin/packages/spectrum-mpi/package.py]: I don't think there's any code in the package that can take advantage of these extra attributes currently.

**Review by becker33 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/spectrum-mpi/package.py (hunk 1 above)
Comment (rblake-llnl) [reply var/spack/repos/builtin/packages/spectrum-mpi/package.py]: Yep, fixed.

File: var/spack/repos/builtin/packages/spectrum-mpi/package.py (hunk 2 above)
Comment (rblake-llnl) [reply var/spack/repos/builtin/packages/spectrum-mpi/package.py]: Commented this out just in case someone wants to resurrect it.

**Review by becker33 — APPROVED** [review]

**Pull request merged**